spec/unit/pops/lookup/context_spec.rb

```diff
@@ -171,13 +171,8 @@ module Lookup
 
       context 'and multiple compilations' do
 
-        before(:each) { Puppet.settings[:environment_timeout] = 'unlimited' }
-        after(:each) do
-          Puppet.settings[:environment_timeout] = 0
-          Puppet.lookup(:environments).clear_all
-        end
-
         it 'will reuse cached_file_data and not call block again' do
+          Puppet.settings[:environment_timeout] = 'unlimited'
```

Comment:
Having the second test default back to `Puppet.settings[:environment_timeout] = 0` instead is what allows the test to pass.  Otherwise, the cache is never refreshed / the file write isn't observed.

If you're sure that the file write should be seen, even if cache doesn't expire, then there's a bug there I believe.